### PR TITLE
Folio 3452:Do not merge

### PR DESCRIPTION
**Pull request by ankitasen-ubmainz**

### CODE REVIEW

**Comment by EthanFreestone** [comment]
This PR is labelled do not merge, so I am converting it to draft. Feel free to change it back if and when it is ready for review by our team

**Comment by EthanFreestone** [comment]
@ankitasen-ubmainz I see you have marked this as ready for review. Please confirm that it is actually for review, since the PR title is still "Do not merge".

**Comment by ankitasen-ubmainz** [comment]
@EthanFreestone Yes, this PR cannot be merged right now since while adding Github Actions ci the pipeline failed with Node v16 but is doing fine with Node v14. This is something the repo owners should help me with, hence the review tag. You can see the exact failure in the actions tab and if you click on the  failed pipelines you can see the ci pipeline fails at yarn test step.

**Comment by EthanFreestone** [comment]
Sorry @ankitasen-ubmainz , I'm not quite sure what you're asking me to help with? In the actions tab I can't see the graphs, it says they will be displayed on the next run. What is there for me to do as the repo maintainer?

**Comment by EthanFreestone** [comment]
As far as I can see, the failures in v16 were outlined by Zak previously and a set of steps were declared which shoudl fix those issues:
https://issues.folio.org/browse/FOLIO-3432

From a glance at the changes it looks like some/all of these are not in place currently. Since we are not the ones pushing this change I had assumed they would be handled by the PR raiser, but if this is something we are expected to do ourselves in order to push this through I would expect to have been informed about that

File: .github/workflows/build-npm-release.yml [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    if : ${{ startsWith(github.ref, 'refs/tags/v') }}
+    env:
+      YARN_TEST_OPTIONS: <options to pass to 'yarn test'>
+      SQ_ROOT_DIR: <replace>
+      COMPILE_TRANSLATION_FILES: <boolean set 'true' or 'false'>
+      PUBLISH_MOD_DESCRIPTOR: <publish mod desc to FOLIO registry (true|false)>
+      FOLIO_NPM_REGISTRY: 'https://repository.folio.org/repository/npm-folio/'
+      FOLIO_NPM_REGISTRY_AUTH: '//repository.folio.org/repository/npm-folio/'
+      FOLIO_MD_REGISTRY: 'https://folio-registry.dev.folio.org'
+      NODEJS_VERSION: '12'
```
Comment (EthanFreestone) [inline .github/workflows/build-npm-release.yml:31]: This line is listed as *should be v16* in [FOLIO-3432](https://issues.folio.org/browse/FOLIO-3432)

**Comment by ankitasen-ubmainz** [comment]
@EthanFreestone The point is with the Node v16 changes, mentioned by @zburke , the pipeline fails but when I bring the version down to Node v14 it works. The workflows were reverted to node v14 and the settings changed for me to test whether the pipeline fails with v14 as well, but strangely the only problem is Node v16. I have updated the build-npm.yml workflow again and as you can see the pipeline fails even after adding the steps.

File: .github/workflows/build-npm-release.yml (hunk 1 above)
Comment (ankitasen-ubmainz) [reply .github/workflows/build-npm-release.yml:31]: This workflow only triggers during relase. The updated workflow is in build-npm.yml. I am testing right now only with that.

**Comment by ankitasen-ubmainz** [comment]
@EthanFreestone if you go to this issue [FOLIO-3540](https://issues.folio.org/browse/FOLIO-3504) you can see a comment from @zburke where he says there were same issues faced during Node v16 migration in ui-users. I am not sure but maybe this information helps? Maybe you could have a look at these?

**Comment by ankitasen-ubmainz** [comment]
@EthanFreestone sure. And do let me know if you need any assistance from me.

**Comment by EthanFreestone** [comment]
@ankitasen-ubmainz I fixed the failing test, so I'm happy to merge on my end if you're happy with this as well :) I presume the release build scripts will also need the changes from that issue at some point though?

**Comment by ankitasen-ubmainz** [comment]
@EthanFreestone thanks a lot. I'll update the remaining workflows and files and merge this.

**Pull request merged**